src/useQuery.ts

```diff
@@ -115,11 +117,15 @@ export function useQuery<TData = any, TVariables = OperationVariables>(
         partial: result.partial,
       };
     },
-    [responseId, observableQuery]
+    [skip, responseId, observableQuery]
```

Comment:
Wondering, if this is really necessary. If a query suddenly becomes skipped, the current result won't change. Why would it?

Reply:
Good question.

Reply:
This prevents infinite loads in suspense mode when skip goes from `false` to `true`, checkout [this test](https://github.com/trojanowski/react-apollo-hooks/pull/42/files#diff-2db9c4351ea93b397a084c99511cc106R654).

<details><summary>with skip in useMemo</summary>

```
  console.log src/useQuery.ts:128
    { skip: true,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }

  console.log src/useQuery.ts:128
    { skip: false,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }

  console.log src/useQuery.ts:128
    { skip: false,
      responseId: 0,
      currentResult:
       { data: { tasks: [Array] },
         error: undefined,
         errors: undefined,
         loading: false,
         partial: false } }
```

</details>


<details><summary>without skip in useMemo</summary>

```
   console.log src/useQuery.ts:128
    { skip: true,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }

  console.log src/useQuery.ts:128
    { skip: false,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }

  console.log src/useQuery.ts:128
    { skip: false,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }

  console.log src/useQuery.ts:128
    { skip: false,
      responseId: 0,
      currentResult:
       { data: {},
         error: undefined,
         errors: undefined,
         loading: true,
         partial: true } }
```

</details>

Reply:
Interesting. I kinda thought that with `useMemo` (and similar) it should be about what variables are actually used within the body of it. Looks like this rule of thumb does not always work 😕